### ISSUE TITLE
automatically initiate a remote add, when necessary

### ISSUE THREAD

**Issue by dustinkirkland** [lxc/lxd#1983]
kirkland@masterbr:~⟫ lxc move unsuffused-mayola loft:
error: unknown remote name: "loft"

lxc should really initiate the "lxc remote add loft:" at this point.

**Comment by stgraber** [comment]
That would only work in cases where the remote name happens to be a valid DNS name, is bound to the default port and either already trusts your client or has a password set.

But yeah, I'm not against the idea, though making this actually happen would need a lot of client rework as there are a LOT of code paths that would need updating and copy/pasting the same logic all over wouldn't be fun. So we have to find the right spot to put it in so that it applies to the Go client binary but not to say client.go as used by other downstream projects (which can't deal with interactive setup).

**Comment by dustinkirkland** [comment]
Right, agreed on all of your assumptions having to be met.  Obviously, it
has to be an IP address, or a resolvable host name, and default port
(8443).  It's fine if you prompt for the password, the first time, to setup
the trust relationship.

This, and the other issue that I just replied to...  Basically, I'm hoping
that lxc would work more like ssh and rsync here.  Those are two old tools,
that *every* system administrator is familiar with.  In both cases, ssh and
rsync, you work with hostnames and/or IP addresses, and there is a default
port (22), or you can specify your own, and keys/certificates are used when
possible, and otherwise, you're prompted for a password.  However, the ssh
and rsync experience would be *weird* if first you had to "ssh add remote
foo foo.example.com" the first time you wanted to "ssh foo.example.com".


:-Dustin
